public/app/features/alerting/unified/plugins/useRulePluginLinkExtensions.ts

```diff
@@ -0,0 +1,76 @@
+import { PluginExtensionPoints } from '@grafana/data';
+import { getPluginLinkExtensions } from '@grafana/runtime';
+import { CombinedRule } from 'app/types/unified-alerting';
+import { PromRuleType } from 'app/types/unified-alerting-dto';
+
+import { getRuleOrigin } from '../utils/rules';
+
+interface BaseRuleExtensionContext {
+  name: string;
+  namespace: string;
+  group: string;
+  expression: string;
+  labels: Record<string, string>;
+}
+
+interface AlertingRuleExtensionContext extends BaseRuleExtensionContext {
+  annotations: Record<string, string>;
+}
+
+interface RecordingRuleExtensionContext extends BaseRuleExtensionContext {}
+
+export function useRulePluginLinkExtension(rule: CombinedRule) {
+  const ruleOrigin = getRuleOrigin(rule);
+  const ruleType = rule.promRule?.type;
+  if (!ruleOrigin || !ruleType) {
+    return [];
+  }
+
+  const { pluginId } = ruleOrigin;
+
+  switch (ruleType) {
+    case PromRuleType.Alerting:
+      return getAlertingRuleActionPluginLinkExtension({ pluginId, rule });
+    case PromRuleType.Recording:
+      return getRecordingRulePluginLinkExtension({ pluginId, rule });
+    default:
+      return [];
+  }
+}
+
+function getAlertingRuleActionPluginLinkExtension({ pluginId, rule }: { pluginId: string; rule: CombinedRule }) {
+  const context: AlertingRuleExtensionContext = {
+    name: rule.name,
+    namespace: rule.namespace.name,
+    group: rule.group.name,
+    expression: rule.query,
+    labels: rule.labels,
+    annotations: rule.annotations,
+  };
+
+  const { extensions } = getPluginLinkExtensions({
+    extensionPointId: PluginExtensionPoints.AlertingRuleAction,
+    context,
+    limitPerPlugin: 3,
+  });
+
+  return extensions.filter((extension) => extension.pluginId === pluginId);
+}
+
+function getRecordingRulePluginLinkExtension({ pluginId, rule }: { pluginId: string; rule: CombinedRule }) {
```

Comment:
A bit confused about the interface here, we want the function to return `{ pluginId, rule }` but `extensions: PluginExtensionLink[]` is what we actually return?

Reply:
That's not the return type, it's the typing for the arguments isn't it?

Reply:
Oh wow I need more coffee, you're absolutely right 🤦

Reply:
Then minor nit: not a big fan of inline type definitions :D

Reply:
This is an inline parameter type. I'll extract to a separate interface so it's more clear